simulator/host_system.go

```diff
@@ -173,6 +173,31 @@ func CreateStandaloneHost(f *Folder, spec types.HostConnectSpec) (*HostSystem, t
 	return host, nil
 }
 
+func (h *HostSystem) DestroyTask(req *types.Destroy_Task) soap.HasFault {
+	if len(h.Vm) > 0 {
```

Comment:
Don't do this here, real VC doesn't fail the creation of tasks, only the running of tasks (which you have right below).